src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -75,29 +84,227 @@ public DarCollectionService(DarCollectionDAO darCollectionDAO, DarCollectionServ
+      }
+    } else {
+      summary.setStatus(DarCollectionStatus.IN_PROCESS.getValue());
+    }
+  }
+
+  private void processDarCollectionSummariesForAdmin(List<DarCollectionSummary> summaries) {
+    //if at least one election is open, show cancel
+    //if at least one non-open/absent election, show open
+    summaries.forEach(s -> {
+      Map<String, Integer> statusCount = new HashMap<>();
+      Map<Integer, Election> elections = s.getElections();
+      if(elections.size() == 0) {
+        s.addAction(DarCollectionActions.OPEN.getValue());
+        s.setStatus(DarCollectionStatus.UNREVIEWED.getValue());
+      } else {
+        elections.values().forEach(e -> {
+          String status = e.getStatus();
+          updateStatusCount(statusCount, status);
+          if(status == ElectionStatus.OPEN.getValue()) {
+            s.addAction(DarCollectionActions.CANCEL.getValue());
+          } else {
+            s.addAction(DarCollectionActions.OPEN.getValue());
+          }
+        });
+        determineCollectionStatus(s, statusCount, s.getDatasetCount(), s.getElections().size());
+      }
+    });
+  }
+
+  private void processDarCollectionDraftsAsSummaries(List<DataAccessRequest> drafts, List<DarCollectionSummary> summaries) {
+    drafts.forEach(d -> {
+      try{
+        DarCollectionSummary summary = new DarCollectionSummary();
+        Date createDate = new Date(d.getData().getCreateDate());
+        String darCode = "DRAFT_DAR_" + new SimpleDateFormat("yyyy-MM-dd")
+          .format(createDate).toString();
+        summary.setDarCode(darCode);
+        summary.setStatus(DarCollectionStatus.DRAFT.getValue());
+        summary.setName(d.getData().getProjectTitle());
+        summary.addAction(DarCollectionActions.RESUME.getValue());
+        summary.addAction(DarCollectionActions.DELETE.getValue());
+        summaries.add(summary);
+      } catch(Exception e) {
+        logger.warn("Error processing draft with id: " + d.getId());
+      }
+    });
+  }
+
+  private void processDarCollectionSummariesForResearcher(List<DarCollectionSummary> summaries) {
+    //if an election exists, cancel does not appear
+    //if there are no elections, review and cancel are present
+    //if the collection is canceled, revise and review is present
+    Map<String, Integer> statusCount = new HashMap<>();
+    summaries.forEach(s -> {
+      Map<Integer, Election> elections = s.getElections();
+      int electionCount = elections.values().size();
+      elections.values().forEach(election -> updateStatusCount(statusCount, election.getStatus()));
+      s.addAction(DarCollectionActions.REVIEW.getValue());
+      //check dar statuses, if they're all canceled show revise (but only if there are no elections)
+      Boolean isCanceled = false;
+      if(electionCount == 0) {
+        Collection<String> darStatuses = s.getDarStatuses().values();
+        isCanceled = darStatuses.size() > 0 && darStatuses.stream()
```

Comment:
`isCanceled` is only used inside the if block so you can instantiate it directly there instead of initializing it here. That way you can avoid an unnecessary variable if `electionCount > 0`
```suggestion
      if(electionCount == 0) {
        Collection<String> darStatuses = s.getDarStatuses().values();
        boolean isCanceled = darStatuses.size() > 0 && darStatuses.stream()
```